src/libraries/Microsoft.Win32.SystemEvents/src/Microsoft/Win32/SystemEvents.cs

```diff
@@ -1248,33 +1255,10 @@ private void WindowThreadProc()
                 {
                     Interop.User32.MSG msg = default(Interop.User32.MSG);
 
-                    bool keepRunning = true;
-
-                    // Blocking on a GetMessage() call prevents the EE from being able to unwind
-                    // this thread properly (e.g. during AppDomainUnload). So, we use PeekMessage()
-                    // and sleep so we always block in managed code instead.
-                    while (keepRunning)
+                    while (Interop.User32.GetMessageW(ref msg, _windowHandle, 0, 0) > 0)
```

Comment:
This seems like a significant behavior change for how this thread works.  Granted it doesn't need to care about AppDomain unload anymore since this is only used on core.  I'm not certain the implications of that.  @dotnet/dotnet-winforms folks would like your feedback.

Reply:
There was a significant change of behavior in .NET 6 (https://github.com/dotnet/runtime/pull/53467) that pushed us to use this code path. Apparently no one really tested it much for the ~4 years that it was there because the code path was simply often not hit in UI applications with their own message loop.

There are multiple issues with the previous version of the message loop. Firstly, it was waking up the loop every 100ms despite no work happening. Secondly, it was not handling the shut down correctly.

I am not fond of the size of the change but the original code is both performing poorly and also functionally incorrect.

Reply:
It's older than 4 years, most of this code was just ported from .NETFramework, so more like 20 years: https://referencesource.microsoft.com/#System/compmod/microsoft/win32/SystemEvents.cs,1556

I like your simplification, I just want the WinForms team to have a look and make sure they don't see a problem with it.